activerecord/test/cases/adapters/postgresql/json_test.rb

```diff
@@ -192,6 +192,20 @@ def test_changes_in_place
 
     assert_equal({ "one" => "two", "three" => "four" }, json.payload)
     assert_not json.changed?
+
+    json.payload = { "three" => "four", "one" => "two" }
```

Comment:
Can we move these tests to a more generic area / make sure the new behavior in `AbstractJson` is being tested properly?